libkbfs/folder_branch_ops.go

```diff
@@ -4027,39 +4100,14 @@ func (fbo *folderBranchOps) UnregisterFromChanges(obs Observer) error {
 
 // notifyBatchLocked sends out a notification for all the ops in md.
 func (fbo *folderBranchOps) notifyBatchLocked(
-	ctx context.Context, lState *lockState, md ImmutableRootMetadata,
-	afterUpdateFn func() error) error {
+	ctx context.Context, lState *lockState, md ImmutableRootMetadata) error {
 	fbo.headLock.AssertLocked(lState)
 
-	switch len(md.data.Changes.Ops) {
-	case 0:
-		panic("Unexpected empty ops change list in notifyBatchLocked")
-	case 1:
-		err := fbo.notifyOneOpLocked(
-			ctx, lState, md.data.Changes.Ops[0], md, false, afterUpdateFn)
-		if err != nil {
-			return err
-		}
-	default:
-		// In a multi-update batch, the first op should be the
-		// resolution op with all the new pointer updates.  It's the
-		// one that gets the afterUpdateFn.
-		if md.data.Changes.Ops[0].(*resolutionOp) == nil {
-			return errors.Errorf("First update in local batch is not a "+
-				"resolutionOp: %T", md.data.Changes.Ops[0])
-		}
-		err := fbo.notifyOneOpLocked(
-			ctx, lState, md.data.Changes.Ops[0], md, false, afterUpdateFn)
+	for _, op := range md.data.Changes.Ops {
+		err := fbo.notifyOneOpLocked(ctx, lState, op, md.ReadOnly(), false)
```

Comment:
Since `syncAllLocked` calls `UpdatePointers` directly, without going through `notifyBatchLocked`, there's no longer any need for complex logic here around the `afterUpdateFn`.